ngraph_bridge/ngraph_utils.cc

```diff
@@ -309,25 +309,55 @@ Status CheckAxisDimInRange(std::vector<int64> axes, size_t rank) {
   return Status::OK();
 }
 
-void NgraphSerialize(const std::string& file_name,
-                     const std::shared_ptr<ngraph::Function>& ng_function) {
-  NGRAPH_VLOG(0) << "Serializing graph to: " << file_name << std::endl;
+Status NgraphSerialize(const std::string& file_name,
+                       const std::shared_ptr<ngraph::Function>& ng_function) {
   int json_indentation = 4;
-  StringToFile(file_name, ngraph::serialize(ng_function, json_indentation));
+  string serialized;
+  if (ng_function == nullptr) {
+    return errors::Internal(
+        "Passed a null pointer as ng function to serialize");
+  }
+  try {
+    ngraph::serialize(ng_function, json_indentation);
+  } catch (...) {
+    return errors::Internal("Failed to serialize ngraph function");
+  }
+  string new_file_name = SanitizeFileName(file_name);
+  NGRAPH_VLOG(0) << "Serializing graph to: " << new_file_name << std::endl;
+  return StringToFile(new_file_name, serialized, false);
```

Comment:
Let StringToFile handle the sanitizing? Pass
StringToFile(new_file_name, serialized, true);